portfolio/src/main/java/com/google/sps/data/TimeRangeGroup.java

```diff
@@ -27,4 +28,11 @@
    * @return the newly modified collection of time ranges.
    */
   public Collection<TimeRange> deleteTimeRange(TimeRange timeRangeToDelete);
+
+  /**
+   * Sorts the collection of time ranges based on given comparator and a boolean value that denotes
+   * if the sorting is in ascending order. The comparator and the boolean value are passed to the
```

Comment:
> The comparator and the boolean value are passed to the constructor of the class.

This sounds pretty dangerous to me. In the `TimeRangeGroupArrayList` below it looks like you let the caller pass in these things in the constructor, so in what scenarios would we need to change the sort order for an already existing `TimeRangeGroup`? I think we should try to treat the sort order as a fundamental immutable property of a `TimeRangeGroup`, because otherwise it'll be really hard to keep track of which sort order it's using at any time.

If you *do* need to change the sort order, I think it would be better to just create a whole new `TimeRangeGroup` with the new sort order.